### DISCUSSION TITLE
Build zenoh-pico on Windows 10

### DISCUSSION THREAD

**Discussion by lanner17** [eclipse-zenoh/roadmap#80] [Ideas and new features]
Hi
Trying to build zenoh-pico on Windows 10 for experimenting. I created a build directory and then ran this command

`cmake -G "Visual Studio 16 2019" ..`

and then 

`cmake --build . --config Release`

It seems to build ok excepts for one repeated warning on almost every file, example

```
C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\winbase.h(9305,5): warning C5105: macro expansion produc
ing 'defined' has undefined behavior [C:\Projects\zenoh\zenoh-pico\build\z_sub_st.vcxproj]
```
It also builds the examples, but I am not able to run the examples. Both `z_pub.exe` and `z_sub.exe` only run to the point where it prints `Opening session...` to the command window. I expected it to progress further than this. For example I expected `z_pub `to print 
`Declaring publisher for `, but it seems to be blocking on either `zp_start_read_task `or `zp_start_lease_task` ?
I also ran a zenoh router, however, it may be not compatible since I build it some time ago.

Thanks

**Comment by Mallets** [comment]
Locking the discussion on roadmap and opening an issue on the more appropriate zenoh-pico repository. 
See: https://github.com/eclipse-zenoh/zenoh-pico/issues/168

**Comment by cguimaraes** [comment]
This has been closed on https://github.com/eclipse-zenoh/zenoh-pico/issues/168
